Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/AppHost/ConsoleWindow.tsx

```diff
@@ -92,6 +95,34 @@ const ConsoleWindow = (props: IProps) => {
     }, [update]);
 
 
+    React.useEffect(() => {
+        if (props.ConsoleURL != undefined && props.ConsoleURL.length > 0) 
+            document.addEventListener("keydown", handleKeyPress, false);
+
+        return () => {
+            document.removeEventListener("keydown", handleKeyPress, false);
+        }
+    }, [props.ConsoleURL])
```

Comment:
Do we need cleanup functions to remove these listeners?

Reply:
Yes we do. When you close the modal it destroys the component. If you leave the listeners attached you create a memory leak